code/modules/fishing/sources/_fish_source.dm

```diff
@@ -166,17 +168,13 @@ GLOBAL_LIST_INIT(specific_fish_icons, generate_specific_fish_icons())
  *
  * For non-fish, it's just the source's fishing difficulty minus the fisherman skill.
  */
-/datum/fish_source/proc/calculate_difficulty(result, obj/item/fishing_rod/rod, mob/fisherman, datum/fishing_challenge/challenge)
+/datum/fish_source/proc/calculate_difficulty(result, obj/item/fishing_rod/rod, mob/fisherman)
 	. = fishing_difficulty
 
 	// Difficulty modifier added by having the Settler quirk
 	if(HAS_TRAIT(fisherman, TRAIT_EXPERT_FISHER))
 		. += EXPERT_FISHER_DIFFICULTY_MOD
 
-	// Difficulty modifier added by the fisher's skill level
-	if(!(challenge?.special_effects & FISHING_MINIGAME_RULE_NO_EXP))
-		. += fisherman.mind?.get_skill_modifier(/datum/skill/fishing, SKILL_VALUE_MODIFIER)
```

Comment:
I've found out this was redundant and already present in the caller. Also never reached because the challenge arg was never set, now removed.